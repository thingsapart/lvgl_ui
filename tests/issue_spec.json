[
  {
    "type": "style",
    "id": "s_debug_style",
    "named": "s_debug",
    "outline_width": 1,
    "outline_color": "#ffeeff",
    "outline_opa": 150,
    "border_width": 1,
    "border_color": "#ffeeff",
    "border_opa": 150,
    "radius": 0
  },
  {
    "//": "This is the object that was causing the instantiation warning.",
    "//": "It defaults to type 'obj' because no 'type' is specified.",
    "//": "It will get a generated name like obj_0 or similar.",
    "children": [
      {
        "type": "label",
        "id": "label_1_in_obj",
        "named": "l_label_1",
        "properties": {
          "text": "TEST"
        }
      }
    ]
  }
]
