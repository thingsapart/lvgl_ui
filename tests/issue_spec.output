CODEGEN_BLOCK: Entering block 0x5612f0d4d110, indent: 1
CODEGEN_BLOCK: Processing IRStmt type: 10 in block 0x5612f0d4d110
CODEGEN_BLOCK: Processing IRStmt type: 8 in block 0x5612f0d4d110
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_outline_width'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_outline_width'
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_outline_color'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_outline_color'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_color_hex'
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_outline_opa'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_outline_opa'
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_border_width'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_border_width'
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_border_color'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_border_color'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_color_hex'
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_border_opa'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_border_opa'
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4d110
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_style_set_radius'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_style_set_radius'
CODEGEN_BLOCK: Exiting block 0x5612f0d4d110
CODEGEN_BLOCK: Entering block 0x5612f0d4ded0, indent: 1
CODEGEN_BLOCK: Processing IRStmt type: 9 in block 0x5612f0d4ded0
CODEGEN_BLOCK: Processing IRStmt type: 8 in block 0x5612f0d4ded0
CODEGEN_BLOCK: Processing IRStmt type: 5 in block 0x5612f0d4ded0
CODEGEN_BLOCK: Entering block 0x5612f0d4e140, indent: 2
CODEGEN_BLOCK: Processing IRStmt type: 9 in block 0x5612f0d4e140
CODEGEN_BLOCK: Processing IRStmt type: 8 in block 0x5612f0d4e140
CODEGEN_BLOCK: Processing IRStmt type: 7 in block 0x5612f0d4e140
CODEGEN_FUNC_CALL_STMT: Generating call for function 'lv_label_set_text'
CODEGEN_EXPR_FUNC_CALL: Processing func_name 'lv_label_set_text'
CODEGEN_BLOCK: Exiting block 0x5612f0d4e140
CODEGEN_BLOCK: Exiting block 0x5612f0d4ded0
// Code generated by LVGL UI Generator

#include "lvgl.h"

// Forward declaration for any registered C pointers you might have from your application.
// Example: extern const lv_font_t my_custom_font;
void create_ui(lv_obj_t* parent) {
    {
        lv_style_t* s_debug = (lv_style_t*)malloc(sizeof(lv_style_t));
        if (s_debug != NULL) {
            memset(s_debug, 0, sizeof(lv_style_t));
            lv_style_init(s_debug);
        } else {
            fprintf(stderr, "Error: Failed to malloc for object %s of type %s\n", "s_debug", "lv_style_t");
        }
        // This is a comment for the style object.
        lv_style_set_outline_width(s_debug, 1);
        lv_style_set_outline_color(s_debug, lv_color_hex(0xFFEEFF));
        lv_style_set_outline_opa(s_debug, 150);
        lv_style_set_border_width(s_debug, 1);
        lv_style_set_border_color(s_debug, lv_color_hex(0xFFEEFF));
        lv_style_set_border_opa(s_debug, 150);
        lv_style_set_radius(s_debug, 0);
    }
    {
        lv_obj_t* obj_0 = lv_obj_create(parent);
        // This is the object that was causing the instantiation warning. It defaults to type 'obj' because no 'type' is specified. It will get a generated name like obj_0 or similar.
        {
            lv_obj_t* l_label_1 = lv_label_create(obj_0);
            // This is a comment for the label object.
            lv_label_set_text(l_label_1, "TEST");
        }
    }
}
