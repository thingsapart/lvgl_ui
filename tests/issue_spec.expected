// Code generated by LVGL UI Generator

#include "lvgl.h"

// Forward declaration for any registered C pointers you might have from your application.
// Example: extern const lv_font_t my_custom_font;
void create_ui(lv_obj_t* parent) {
    {
        lv_style_t* s_debug = (lv_style_t*)malloc(sizeof(lv_style_t));
        if (s_debug != NULL) {
            memset(s_debug, 0, sizeof(lv_style_t));
            lv_style_init(s_debug);
        } else {
            fprintf(stderr, "Error: Failed to malloc for object %s of type %s\n", "s_debug", "lv_style_t");
        }
        // This is a comment for the style object.
        lv_style_set_outline_width(s_debug, 1);
        lv_style_set_outline_color(s_debug, lv_color_hex(0xFFEEFF));
        lv_style_set_outline_opa(s_debug, 150);
        lv_style_set_border_width(s_debug, 1);
        lv_style_set_border_color(s_debug, lv_color_hex(0xFFEEFF));
        lv_style_set_border_opa(s_debug, 150);
        lv_style_set_radius(s_debug, 0);
        registry_add_pointer(ui_registry, s_debug, "s_debug_style", "style");
    }
    {
        lv_obj_t* obj_0 = lv_obj_create(parent);
        // This is the object that was causing the instantiation warning. It defaults to type 'obj' because no 'type' is specified. It will get a generated name like obj_0 or similar.
        {
            lv_obj_t* l_label_1 = lv_label_create(obj_0);
            // This is a comment for the label object.
            lv_label_set_text(l_label_1, "TEST");
            registry_add_pointer(ui_registry, l_label_1, "label_1_in_obj", "label");
        }
    }
}
